catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/transformer/RegistryServiceConverter.java

```diff
@@ -0,0 +1,441 @@
+     * @param meta
+     */
+    private void readOrgInfo(HierarchicalStreamReader reader, RegistryMetacardImpl meta) {
+
+        reader.moveDown();
+        reader.moveDown();
+        reader.moveDown();
+
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            switch (reader.getNodeName()) {
+
+            case "rim:Name":
+                meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_NAME, readValue(reader));
+                break;
+            case "rim:EmailAddress":
+                meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_EMAIL,
+                        reader.getAttribute("address"));
+                break;
+            case "rim:TelephoneNumber":
+                meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_PHONE_NUMBER,
+                        readPhoneNumber(reader));
+                break;
+            case "rim:Address":
+                meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_ADDRESS,
+                        readAddress(reader));
+                break;
+            }
+
+            reader.moveUp();
+        }
+
+        reader.moveUp();
+        reader.moveUp();
+        reader.moveUp();
+    }
+
+    String readValue(HierarchicalStreamReader reader) {
+        reader.moveDown();
+        String value = reader.getAttribute("value");
+        reader.moveUp();
+        return value;
+    }
+
+    /**
+     * Read gml and return a wkt.
+     *
+     * @param reader
+     * @return
+     */
+    private String readGml(HierarchicalStreamReader reader) {
+
+        reader.moveDown();
+        reader.moveDown();
+        reader.moveDown();
+
+        // buffer up the current gml node and all its children
+        Writer out = new StringWriter();
+        HierarchicalStreamWriter writer = new CompactWriter(out);
+        HierarchicalStreamCopier copier = new HierarchicalStreamCopier();
+        copier.copy(reader, writer);
+        String xml = out.toString();
+
+        xml = xml.replaceFirst(">", " xmlns:gml=\"http://www.opengis.net/gml\">");
+
+        reader.moveUp();
+        reader.moveUp();
+        reader.moveUp();
+
+        return gml3ToWkt.convert(xml);
+    }
+
+    /**
+     * Read classification info from a rim:Classification node and attach to metacard.
+     *
+     * @param reader
+     * @param meta
+     */
+    private void readClassification(HierarchicalStreamReader reader, MetacardImpl meta) {
+        while (reader.hasMoreChildren()) {
+            reader.moveDown();
+
+            switch (reader.getNodeName()) {
+
+            case "rim:Slot":
+                String name = reader.getAttribute("name");
+                switch (name) {
+                case "location":
+                    meta.setAttribute(Metacard.GEOGRAPHY, readGml(reader));
+                    break;
+                default:
+                    String key = CLASSIFICATIONS.get(name);
+                    if (key != null) {
```

Comment:
Consider pulling these nested `if` statements in the `switch` statement into their own method.